vaccine_feed_ingest/runners/az/arcgis/normalize.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+
+import json
+import logging
+import os
+import pathlib
+import re
+import sys
+from datetime import datetime
+from typing import List, Optional
+
+# import schema
+site_dir = pathlib.Path(__file__).parent
+state_dir = site_dir.parent
+runner_dir = state_dir.parent
+root_dir = runner_dir.parent
+sys.path.append(str(root_dir))
+
+from schema import schema  # noqa: E402
+
+# Configure logger
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s:%(name)s:%(message)s",
+    datefmt="%m/%d/%Y %H:%M:%S",
+)
+logger = logging.getLogger("az/arcgis/normalize.py")
+
+output_dir = pathlib.Path(sys.argv[1])
+input_dir = pathlib.Path(sys.argv[2])
+
+json_filepaths = input_dir.glob("*.ndjson")
+
+parsed_at_timestamp = datetime.utcnow().isoformat()
+
+
+def _get_id(site: dict) -> str:
+    data_id = site["attributes"]["globalid"]
+
+    # Could parse these from directory traversal, but do not for now to avoid
+    # accidental mutation.
+    site = "arcgis"
+    runner = "az"
+
+    # Could parse these from the input file name, but do not for now to avoid
+    # accidental mutation.
+    arcgis = "128ead309d754558ad81bccd99188dc9"
+    layer = 0
+
+    return f"{runner}:{site}:{arcgis}_{layer}:{data_id}"
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    if site["attributes"]["prereg_phone"]:
+        sourcePhone = re.sub("[^0-9]", "", site["attributes"]["prereg_phone"])
+        if len(sourcePhone) == 11:
+            sourcePhone = sourcePhone[1:]
+        phone = f"({sourcePhone[0:3]}) {sourcePhone[3:6]}-{sourcePhone[6:]}"
+        contacts.append(schema.Contact(phone=phone))
+
+    if site["attributes"]["prereg_website"]:
+        contacts.append(schema.Contact(website=site["attributes"]["prereg_website"]))
+
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+def _get_languages(site: dict) -> Optional[List[str]]:
+    return {
+        None: None,
+        "Yes": ["en", "es"],
+        "No": ["en"]
+    }[site["attributes"]["spanish_staff_y_n"]]
+
+
+def _get_opening_dates(site: dict) -> Optional[List[schema.OpenDate]]:
+    if site["attributes"]["begindate"] is None or site["attributes"]["enddate"] is None:
+        return None
```

Comment:
It is okay to add _just_ an `opens` date or _just_ a `closes` date in the normalized schema (we commonly see sites which opened in the past - so might not have a date recorded, and sites which don't have a close date yet assigned but we don't want to miss later).

Could we loosen this restriction?

```suggestion
    if site["attributes"]["begindate"] is None and site["attributes"]["enddate"] is None:
        return None
```

Reply:
Updated to allow having only one of `opens` and `closes`.